app/code/Magento/SendFriend/Controller/Product.php

```diff
@@ -5,58 +5,57 @@
  */
 namespace Magento\SendFriend\Controller;
 
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Catalog\Model\Product as ModelProduct;
+use Magento\Catalog\Model\Session as CatalogSession;
+use Magento\Customer\Model\Session as CustomerSession;
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\ResponseInterface;
+use Magento\Framework\Data\Form\FormKey\Validator as FormKeyValidator;
 use Magento\Framework\Exception\NotFoundException;
 use Magento\Framework\App\RequestInterface;
 use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Registry;
+use Magento\SendFriend\Helper\Data as SendFriendHelper;
+use Magento\SendFriend\Model\SendFriend;
 
 /**
  * Email to a Friend Product Controller
  *
  * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
  */
-abstract class Product extends \Magento\Framework\App\Action\Action
+abstract class Product extends Action
 {
     /**
      * Core registry
      *
-     * @var \Magento\Framework\Registry
+     * @var Registry
      */
     protected $_coreRegistry = null;
 
     /**
-     * @var \Magento\Framework\Data\Form\FormKey\Validator
+     * @var FormKeyValidator
      */
     protected $_formKeyValidator;
 
     /**
-     * @var \Magento\SendFriend\Model\SendFriend
-     */
-    protected $sendFriend;
-
-    /**
-     * @var \Magento\Catalog\Api\ProductRepositoryInterface
-     */
-    protected $productRepository;
-
-    /**
-     * @param \Magento\Framework\App\Action\Context $context
-     * @param \Magento\Framework\Registry $coreRegistry
-     * @param \Magento\Framework\Data\Form\FormKey\Validator $formKeyValidator
-     * @param \Magento\SendFriend\Model\SendFriend $sendFriend
-     * @param \Magento\Catalog\Api\ProductRepositoryInterface $productRepository
+     * @param Context $context
+     * @param Registry $coreRegistry
+     * @param FormKeyValidator $formKeyValidator
+     * @param SendFriend $sendFriend
+     * @param ProductRepositoryInterface $productRepository
      */
     public function __construct(
-        \Magento\Framework\App\Action\Context $context,
-        \Magento\Framework\Registry $coreRegistry,
-        \Magento\Framework\Data\Form\FormKey\Validator $formKeyValidator,
-        \Magento\SendFriend\Model\SendFriend $sendFriend,
-        \Magento\Catalog\Api\ProductRepositoryInterface $productRepository
+        Context $context,
+        Registry $coreRegistry,
+        FormKeyValidator $formKeyValidator,
```

Comment:
Why not promote these properties?
```suggestion
        protected readonly Registry $_coreRegistry,
        protected readonly FormKeyValidator $_formKeyValidator,
```